pkg/cli/dev.go

```diff
@@ -59,7 +64,27 @@ func (s *Dev) Run(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	imageSource := imagesource.NewImageSource(s.client.AcornConfigFile(), s.File, s.ArgsFile, args, nil, z.Dereference(s.AutoUpgrade))
+	var imageSource imagesource.ImageSource
+	if !s.Clone {
+		imageSource = imagesource.NewImageSource(s.client.AcornConfigFile(), s.File, s.ArgsFile, args, nil, z.Dereference(s.AutoUpgrade))
+	} else if s.Name == "" {
+		return fmt.Errorf("clone option must be used when running dev on an existing app")
```

Comment:
nit: the current wording of this makes it sound like `--clone` is necessary

```suggestion
		return fmt.Errorf("clone option can only be used when running dev on an existing app")
```